index.md

```diff
@@ -8,6 +8,469 @@ This is some welcome text.
+    if foo < 0:
+        return False
+    else:
+        return True
+```
+
+### C0103 Invalid Name {#C0103}
+
+This error occurs when an invalid name is used as a variable name. This type of 
+error is a sign that the variable names should be modified to be of the form: a 
+lowercase letter followed by a lowercase letter or digit. The minimum number of 
+characters should be 2 and the maximum should be 30.
+
+```python
+def is_equal(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    Ex = 5
+    return obj == Ex
+```
+
+### C0111 Missing Docstring {#C0111}
+
+This error occurs when a module, function, class or method has no docstring. 
+Some special methods like __init__ don't require a docstring. Students are 
+expected to write docstrings for every module, function, class and method (maybe
+ except the special methods with '__').
+
+```python
+def is_true(obj):
+    if obj is False:
+        return False
+    else:
+        return True
+```
+
+### C0112 Empty Docstring {#C0112}
+
+This error occurs when a module, function, class or method has an empty 
+docstring. Students are expected to write complete docstrings for every module, 
+function, class and method.
+
+```python
+def is_true(obj):
+    """
+    """
+    if obj is False:
+        return False
+    else:
+        return True
+```
+
+### C0113 Unneeded not {#C0113}
+
+### C0121 Singleton comparison {#C0121}
+
+This is an error that occurs when an expression is compared to singleton values 
+like True, False or None. This type of error indicates incorrect comparison 
+tests, so please check your code carefully.
+
+```python
+def is_true():
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    temp = 5
+    if temp == True:
+        return False
+    else:
+        return True
+```
+
+### C0122 Misplaced comparison constant {#C0122}
+
+This error occurs when a constant is placed on the left side of a comparison. 
+It is usually a better practice to place it in the right side of the comparison.
+
+```python
+def pos(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    if 0 > obj:
+        return False
+    else:
+        return True
+```
+
+### C0123 Unidiomatic type check {#C0123}
+
+This error occurs when type() is used instead of isinstance() for a type check. Students are expected to use isinstance(x, Y) instead of type(x) == Y. The example below demonstrates the same.
+
+```python
+def is_int(obj):
+    """Check is the given object is of type 'int'.
+    
+    @type obj: object
+    @rtype: bool
+    """
+    return type(obj) == "<type 'int'>"
+```
+
+The above can be modified to:
+
+```python
+def is_int(obj):
+    """Check is the given object is of type 'int'.
+    
+    @type obj: object
+    @rtype: bool
+    """
+    return isinstance(obj, int)
+```
+
+### E0101 Return in init {#E0101}
+
+This error occurs when a return statement is used in the __init__ method. 
+Students are expected to not return anything in the __init__ method.
```

Comment:
Okay, don't use the phrase "Students are expected to" too much. Instead, explain why you don't need to return anything in the init method.